### PR TITLE
Pass `currentState` to filter.apply()

### PR DESCRIPTION
**Pull request by finscn**
see https://github.com/pixijs/pixi.js/issues/3317 

If  pass ```currentState``` to filter.apply() , users could get more informations  (e.g. get current sprite by ```currentSprite = currentState.target```).
It's very useful when users create custom filters.

And the ```currentState``` is the last argument , and pass it won't break any filter already existed.

So I think this PR is safe and useful.

### CODE REVIEW

**Comment by finscn** [comment]
I've add the parameter  in `apply` of Filter.js , but eslint failed. because it doesn't be used.

**Comment by finscn** [comment]
@GoodBoyDigital   ,  I've updated `apply` function parameters on the `Filter` class .
Review please.

**Comment by bigtimebuddy** [comment]
@GoodBoyDigital any follow-up here? 

@finscn could you resolve conflicts?

**Comment by finscn** [comment]
@GoodBoyDigital , I've updated apply function parameters on the Filter class .
Review please.  Is there any other problem ?

**Comment by finscn** [comment]
Is there any news about this PR ?

**Comment by finscn** [comment]
Don't forget this PR ,please.

 @GoodBoyDigital  , you  requested changes , I've changed.

File: src/core/renderers/webgl/filters/Filter.js [hunk 1]
```diff
@@ -96,12 +96,24 @@ export default class Filter
...
+        /**
+         * There are some useful properties in currentState :
+         *     filters
+         *     target
+         *     renderTarget
+         *     resolution
+         *     sourceFrame
+         *     destinationFrame
+         */
+        const currentDisplayObject = currentState.target; // eslint-disable-line no-unused-vars
```
Comment (englercj) [inline src/core/renderers/webgl/filters/Filter.js]: Why does this line exist?

Comment (finscn) [inline src/core/renderers/webgl/filters/Filter.js]: If no this line , eslint will report error

Comment (finscn) [inline src/core/renderers/webgl/filters/Filter.js]: @englercj , I've changed the code.

**Comment by englercj** [comment]
lg2m, @GoodBoyDigital good to you?

**Comment by finscn** [comment]
@GoodBoyDigital  , is there any question about this PR ?

**Pull request merged**